Update Index sizing guidelines in docs after Autosharding (#7151)
elastic/docs-content: deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings.md

| **[LogsDB](/manage-data/data-store/data-streams/logs-data-stream.md)** | Optional | ✅ | - Enabled by default <br>- Cannot be disabled |
| **SIEM capabilities** | ✅ | ✅ | Core functionality supported |

## Serverless index sizing and resource limits [elasticsearch-differences-serverless-index-size]

> Elasticsearch uses sharding to distribute the data in your index across the cluster - this helps to support growing data volumes and load. In {{serverless-full}}, the sharding is fully managed for you. You don’t need to think about choosing the optimal shard count for an index because the sharding will automatically adjust to suit your needs. That said, there are still some guidelines to share: these are best practices that tend to give the best results over the long term. You should design your indexing and data lifecycle strategy with the size and growth of your data in mind.

Reviewer: Suggesting a small edit for minimalism and to avoid repetition:

```suggestion
Elasticsearch uses sharding to distribute the data in your index across the cluster - this helps support growing data volumes and load. In {{serverless-full}}, the sharding is fully managed for you and will automatically adjust to suit your needs.

The following best practices will help you build an indexing and data lifecycle strategy that scales with your data over time:

```

Author: Looks good, thanks.

> :

Reviewer: Removing the colon here as it's included in the preceding paragraph.
```suggestion

```

>  * Large numbers of very small (or empty) indices should be avoided whenever possible. Each index has a resource cost within your project and the service must scale your project capacity to accommodate. If you recognize this pattern in your design, consider options for organizing your data into fewer indices.

Reviewer: ```suggestion
 * Avoid large numbers of very small or empty indices where possible. Each index has a resource cost within your project and the service must scale its capacity to accommodate. If your design follows this pattern, consider organizing your data into fewer indices.
```

This looks very much like the [indexing strategies](https://www.elastic.co/docs/deploy-manage/cloud-organization/billing/elasticsearch-billing-dimensions#elasticsearch-billing-indexing-strategies) we describe in the billing docs. See the **index count and size** list item.

Should these be aligned more?

Author: I think it's alright for these to be separate guidelines. The guidelines we are describing here have to do with performance, not cost.

Reviewer: Fair point, makes sense. 
Will I go ahead and apply the editing suggestions as they're not really tied into cost guidelines or anything?

>  * At the other end of the spectrum, a design where each index can grow to many terabytes might also ultimately result in performance trade-offs - consider creating multiple smaller indices that you can query using an [alias](/manage-data/data-store/aliases.md) or naming pattern.

Reviewer: Maybe we can say something like:

```suggestion
 * Indices that grow too large can introduce performance trade-offs. Consider creating multiple smaller indices you can query using an [alias](/manage-data/data-store/aliases.md) or naming pattern.
```

Author: I kind of like the original wording because it drives home the distinction between a few very large indices and a large number of small indices. 
I also want to keep the wording about "grow to many terabytes" to concretize a bit what we mean by large indices.

Reviewer: Understood, how about somewhere in the middle? Does this sound any better:

```suggestion
 * By contrast, indices that grow to multiple terabytes can introduce performance trade-offs. Consider creating multiple smaller indices you can query using an [alias](/manage-data/data-store/aliases.md) or naming pattern.
```

>  * Use a data stream instead of a single index when appropriate, and take advantage of data stream lifecycle to manage data retention. Using date-based naming patterns with data streams is generally undesirable. It’s better to let the [data stream lifecycle](/manage-data/lifecycle/data-stream.md) take care of aging out data that is no longer relevant.

Reviewer: Can we define when is appropriate? is it when we deal with time-series data?

```suggestion
 * When your data is time-series or append-only, use a data stream instead of a single index and take advantage of the [data stream lifecycle](/manage-data/lifecycle/data-stream.md) to manage data retention. Avoid using date-based naming patterns with data streams and let the lifecycle age out data automatically.
```

Author: I have added a link here to a page that talks about when to use data streams.

{{serverless-full}} applies the following project-level limit to ensure reliable performance and stability.

| Limit | Value | Adjustable |
| :--- | :--- | :--- |
| Number of indices per project | 15,000 | Yes |

> Adjustable limits can be increased by request, while others are fixed. To request a limit increase, open a support case, and include your preferred new value and a brief description of your use case. Providing meaningful details around your use case and desired outcome ensures that Elastic can make recommendations that best suit your workload.

Reviewer: For clarity, I'd keep the mention of **index** as it's the index limits that are adjustable. 

```suggestion
Index limits can be increased by request, while others are fixed. To request a limit increase, open a support case, and include your preferred new value and a brief description of your use case. Providing meaningful details around your use case and desired outcome ensures that Elastic can make recommendations that best suit your workload.
```

Author: Changed this back to original wording.

| **[LogsDB](/manage-data/data-store/data-streams/logs-data-stream.md)** | Optional | ✅ | - Enabled by default <br>- Cannot be disabled |
| **SIEM capabilities** | ✅ | ✅ | Core functionality supported |

## Elasticsearch index sizing guidelines [elasticsearch-differences-serverless-index-size]

> To ensure optimal performance in Serverless Elasticsearch projects, follow these sizing recommendations.

Reviewer 2: the heading and therefore body might be wrong here. I assume that this index sizing guideline applies to all projects equally, not just ES ones?

> If you created your index after **June 1, 2026**, your index can grow upto 4.8TB without any performance impact.

Reviewer 3: Is 4.8TB the max index size that has been tested across all use-cases w/o any performance impact?

Author: Thanks for reviewing Yuvi. 
I am actually not sure if we have specific tests with large number of shards. Let me check and get back.

The original guidelines calculate the maximum index size based on 100 GB shards and default shard count per project. This is because performance is expected to degrade when shards grow larger than 100 GB.
With Autosharding, we are able to increase the number of shards in the index once the shards get larger than 100 GB. That means we can support a larger index while still keeping the shard size to 100 GB. We currently have an upper limit on the number of shards an index can auto shard upto (=48). This is just because autosharding is a new feature and we wanted to prevent against runaway autosharding. So the 4.8TB number is just 48 x 100 GB.
We believe that autoscaling will take care of distributing the shards across nodes as needed, as the index grows.

Reviewer 3: Thanks Ankita for the comment. Let's confirm if we have tests that can align with the statement (without any performance impact) so that we can say it with confidence.

Reviewer 4: I wonder if we want to be this specific. 48 shards is just a limit of autosharding feature and i can imagine us manually resharding to 96 (and beyond?) on a case by case basis. AFAIK there is no technical limitation here?

Author: I agree @lkts. We can round that up to ~5 TB. I do mention that the limit can be raised further via overrides. We could reword that as well to say something like "This is a soft limit and can be raised on a case-by-case basis". 
I wanted to see if Yuvi agrees with this doc change and how much information we want to give the customer.
I talked to Jason about performance tests and he agrees that it will be good to set a baseline for 6 shards x 100 GB indices to compare against larger 48 x 100 GB indices. I am working with him to run some tests.

Reviewer 4: I would maybe frame this as "There is no limit and we have confirmed internally that 5 TB definitely works. Work with support if your use case exceeds that.". What do you think (obviously wording can be way nicer)?

> If you created your index after **June 1, 2026**, your index can grow upto 4.8TB without any performance impact.

Reviewer 3: Are there any caveats to June 1 2026 or we can saw explicitly that all indices after June 1 2026 across all project types (O11y, Security, ES3) benefit from this.

Author: The June 1, 2026 date comes from the fact that the minimum required index version `SHARD_OBLIVIOUS_SLICING` was committed on May 22 and was picked up in production by June 1. 

I can't think of any other caveats.

> That limit can be raised further via overrides.

Reviewer 4: I am not sure we should advertise overrides in public documentation. Is there precedence for this?

Reviewer 3: So "overrides" is not a public facing word that we use; so we cannot use that. 

We have precedence on the index limit (as an example) where we mention that there are limits that can be increased.

Verbiage from the [public doc](https://www.elastic.co/docs/deploy-manage/deploy/elastic-cloud/differences-from-other-elasticsearch-offerings#index-and-resource-limits):

> The index limit is adjustable and can be increased by request, while others are fixed. To request a limit increase, open a support case, and include your preferred new value and a brief description of your use case. Providing meaningful details around your use case and desired outcome ensures that Elastic can make recommendations that best suit your workload.

So similar to that, if we are mentioning a limit (that can be increased), we can use similar verbiage (or just combine within existing verbiage that it also applies to this limit).

Author: Agreed. Perhaps we can say "This is a soft limit and can be raised on a case-by-case basis" ?

> That limit can be raised further via certain project overrides.

Reviewer 3: Is there a knob that we control internally that we can use to increase the index size above 4.8TB without any performance impact? If yes, do we know the max size?

Author: The knob is `AUTO_RESHARD_MAX_SHARDS_SETTING`, but again, I am not sure about tests.

Reviewer 3: This seems to be an internal knob that we don't expose to customers, right?

Author: That's correct, my understanding is that we don't want to expose shard level details to serverless customers.

> If you created your index before **June 1, 2026**, follow these recommendations according to project type:

Reviewer 2: can we tell them to reindex if the size becomes an issue? will they automatically get the benefit for models where the index is abstracted, like data streams?

| **[LogsDB](/manage-data/data-store/data-streams/logs-data-stream.md)** | Optional | ✅ | - Enabled by default <br>- Cannot be disabled |
| **SIEM capabilities** | ✅ | ✅ | Core functionality supported |

## Serverless index sizing and resource limits [elasticsearch-differences-serverless-index-size]

> Elasticsearch uses sharding to distribute the data in your index across the cluster - this helps support growing data volumes and load. In {{serverless-full}}, the sharding is fully managed for you and will automatically adjust to suit your needs. 

Reviewer 5: ```suggestion
## Serverless index sizing and resource limits [elasticsearch-differences-serverless-index-size]

{{es}} uses sharding to distribute the data in your index across the cluster, which helps support growing data volumes and load. In {{serverless-full}}, sharding is fully managed for you and automatically adjusts to suit your needs.
```

> The following best practices will help you build an indexing and data lifecycle strategy that scales with your data over time:

Reviewer 5: ```suggestion
These best practices help you build an indexing and data lifecycle strategy that scales with your data over time:
```

 * Large numbers of very small (or empty) indices should be avoided whenever possible. Each index has a resource cost within your project and the service must scale your project capacity to accommodate. If you recognize this pattern in your design, consider options for organizing your data into fewer indices.

>  * At the other end of the spectrum, a design where each index can grow to many terabytes might also ultimately result in performance trade-offs. Consider creating multiple smaller indices that you can query using an [alias](/manage-data/data-store/aliases.md) or naming pattern.

Reviewer 5: ```suggestion
* Avoid designs where each index grows to many terabytes, which can result in performance trade-offs. Consider creating multiple smaller indices that you can query using an [alias](/manage-data/data-store/aliases.md) or naming pattern.
```

>  * Use a data stream instead of a single index when [appropriate](/manage-data/data-store/data-streams.md#should-you-use-a-data-stream), and take advantage of the [data stream lifecycle](/manage-data/lifecycle/data-stream.md) to manage data retention. Avoid using date-based naming patterns with data streams and let the lifecycle age out data automatically.

Reviewer 5: ```suggestion
* Use a data stream instead of a single index when [appropriate](/manage-data/data-store/data-streams.md#should-you-use-a-data-stream), and take advantage of the [data stream lifecycle](/manage-data/lifecycle/data-stream.md) to manage data retention. Avoid date-based naming patterns with data streams, and let the lifecycle age out data automatically.
```
